client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -173,6 +178,72 @@ const StatsPurchasePage = ( {
 						) }
 					</>
 				) }
+				{ ! isLoading && isTypeDetectionEnabled && (
+					<>
+						{
+							// a plan is owned - show a notice page for a commercial plan
+							// or a banner for a free and PWYW plans with a wizard below the notice
+							! showPurchasPage && (
+								<StatsPurchaseNoticePage
+									siteSlug={ siteSlug }
+									isCommercialOwned={ isCommercialOwned }
+									isFreeOwned={ isFreeOwned }
+									isPWYWOwned={ isPWYWOwned }
+								/>
+							)
+						}
+						{
+							// blog doesn't have any plan but is not categorised as either personal or commectial - show old purchase wizard
+							( isFreeOwned || isPWYWOwned || ( showPurchasPage && ! showSinglePurchasPage ) ) && (
+								<StatsPurchaseWizard
+									siteSlug={ siteSlug }
+									commercialProduct={ commercialProduct }
+									maxSliderPrice={ maxSliderPrice ?? 10 }
+									pwywProduct={ pwywProduct }
+									siteId={ siteId }
+									redirectUri={ query.redirect_uri ?? '' }
+									from={ query.from ?? '' }
+									disableFreeProduct={ isFreeOwned || isCommercialOwned || isPWYWOwned }
+									initialStep={ initialStep }
+									initialSiteType={ initialSiteType }
+								/>
+							)
+						}
+						{ showPurchasPage && showSinglePurchasPage && (
```

Comment:
```suggestion
						{ showSinglePurchasPage && (
```

We don't need to check `showPurchasPage` here since `showSinglePurchasPage` has included the checking.